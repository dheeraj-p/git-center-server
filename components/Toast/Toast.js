import ToastStyles from './Toast.style';

export default function Toast({ message, isVisible, onClose }) {
  const display = isVisible ? 'visible' : 'hidden';
  if (isVisible) setTimeout(onClose, 3000);

  return (
    <div className={`d-flex justify-content-center toast-container ${display}`}>
      <span className="toast-message text-light">{message}</span>
      <style jsx>{ToastStyles}</style>
    </div>
  );
}
