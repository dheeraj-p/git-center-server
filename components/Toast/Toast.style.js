import css from 'styled-jsx/css';
import colors from '../../styles/colors';

const ToastStyles = css`
  .toast-container {
    position: -webkit-fixed;
    position: fixed;
    width: 100vw;
    bottom: 4vh;
    left: 0;
    z-index: 1000;
    transition: visibility 0.5s ease-out, opacity 0.5s ease-out;
  }

  .visible {
    visibility: visible;
    opacity: 1;
  }

  .hidden{
    visibility: hidden;
    opacity: 0;
  }

  .toast-message {
    background-color: ${colors.colorBackgroundDark};
    padding: 6px 16px;
    font-weight: lighter;
    border-radius: 4px;
  }
`;

export default ToastStyles;
