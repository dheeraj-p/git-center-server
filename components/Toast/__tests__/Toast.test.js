import { shallow } from 'enzyme';
import Toast from '../../Toast/Toast';

describe('Toast', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Toast isVisible={true} message={'Sample toast'} />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should have classname `visible` when Toast is visible', () => {
    const wrapper = shallow(
      <Toast isVisible={true} message={'Sample toast'} />
    );

    expect(wrapper.find('.toast-container').hasClass('visible')).toBeTruthy();
  });

  it('should have classname `hidden` when Toast is hidden', () => {
    const wrapper = shallow(
      <Toast isVisible={false} message={'Sample toast'} />
    );

    expect(wrapper.find('.toast-container').hasClass('hidden')).toBeTruthy();
  });

  it('should be hidden after 3000ms', () => {
    jest.clearAllMocks();
    const onClose = jest.fn();
    shallow(
      <Toast isVisible={true} message={'Sample toast'} onClose={onClose} />
    );

    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(onClose, 3000);
  });
});
