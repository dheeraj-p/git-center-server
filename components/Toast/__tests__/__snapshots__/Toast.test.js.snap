// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render properly 1`] = `
"<div className=\\"jsx-3000152423 d-flex justify-content-center toast-container visible\\">
  <span className=\\"jsx-3000152423 toast-message text-light\\">
    Sample toast
  </span>
  <JSXStyle id=\\"3000152423\\">
    .
    t
    o
    a
    s
    t
    -
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    3
    0
    0
    0
    1
    5
    2
    4
    2
    3
    {
    p
    o
    s
    i
    t
    i
    o
    n
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    i
    x
    e
    d
    ;
    p
    o
    s
    i
    t
    i
    o
    n
    :
    f
    i
    x
    e
    d
    ;
    w
    i
    d
    t
    h
    :
    1
    0
    0
    v
    w
    ;
    b
    o
    t
    t
    o
    m
    :
    4
    v
    h
    ;
    l
    e
    f
    t
    :
    0
    ;
    z
    -
    i
    n
    d
    e
    x
    :
    1
    0
    0
    0
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    t
    r
    a
    n
    s
    i
    t
    i
    o
    n
    :
    v
    i
    s
    i
    b
    i
    l
    i
    t
    y
     
    0
    .
    5
    s
     
    e
    a
    s
    e
    -
    o
    u
    t
    ,
    o
    p
    a
    c
    i
    t
    y
     
    0
    .
    5
    s
     
    e
    a
    s
    e
    -
    o
    u
    t
    ;
    t
    r
    a
    n
    s
    i
    t
    i
    o
    n
    :
    v
    i
    s
    i
    b
    i
    l
    i
    t
    y
     
    0
    .
    5
    s
     
    e
    a
    s
    e
    -
    o
    u
    t
    ,
    o
    p
    a
    c
    i
    t
    y
     
    0
    .
    5
    s
     
    e
    a
    s
    e
    -
    o
    u
    t
    ;
    }
    .
    v
    i
    s
    i
    b
    l
    e
    .
    j
    s
    x
    -
    3
    0
    0
    0
    1
    5
    2
    4
    2
    3
    {
    v
    i
    s
    i
    b
    i
    l
    i
    t
    y
    :
    v
    i
    s
    i
    b
    l
    e
    ;
    o
    p
    a
    c
    i
    t
    y
    :
    1
    ;
    }
    .
    h
    i
    d
    d
    e
    n
    .
    j
    s
    x
    -
    3
    0
    0
    0
    1
    5
    2
    4
    2
    3
    {
    v
    i
    s
    i
    b
    i
    l
    i
    t
    y
    :
    h
    i
    d
    d
    e
    n
    ;
    o
    p
    a
    c
    i
    t
    y
    :
    0
    ;
    }
    .
    t
    o
    a
    s
    t
    -
    m
    e
    s
    s
    a
    g
    e
    .
    j
    s
    x
    -
    3
    0
    0
    0
    1
    5
    2
    4
    2
    3
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    -
    c
    o
    l
    o
    r
    :
    #
    2
    c
    2
    8
    2
    8
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    6
    p
    x
     
    1
    6
    p
    x
    ;
    f
    o
    n
    t
    -
    w
    e
    i
    g
    h
    t
    :
    l
    i
    g
    h
    t
    e
    r
    ;
    b
    o
    r
    d
    e
    r
    -
    r
    a
    d
    i
    u
    s
    :
    4
    p
    x
    ;
    }
  </JSXStyle>
</div>"
`;
